### DISCUSSION TITLE
join spelled characters into words

### DISCUSSION THREAD

**Discussion by diegoasua** [deepgram/community#605] [General help] [answered]
Sometimes when dictating a difficult word, such an uncommon surname or the name of a street one might choose to spell it instead. When this happens Deepgram transcribes every letter separated by a space instead of combining into a single word. For example, if one spells "d u r a n" the expected transcription should be "Duran". Is there any settings or flags that would enable parsing automatically? Or a suggestion how to parse spelled characters?

Using streaming nova-2 
```javascript
const liveTranscription = deepgram.listen.live({
    language: 'en',
    smart_format: false,
    filler_words: false,
    punctuate: true,
    numerals: false,
    // interim_results must be true to use utterance_end_ms
    interim_results: true,
    utterance_end_ms: 1500,
    model: "nova-2",
    encoding: "linear16",
    sample_rate: 16000,
    channels: 1,
    diarize: true,
    // ms to detect with VAD
    endpointing: 1000,
});
```

**Comment by team-deepgram** [comment]
Thanks for asking your question about Deepgram! If you didn't already include it in your post, please be sure to add as much detail as possible so we can assist you efficiently, such as:
- The `request_id` if you have a question about your requests or transcription responses.
- The features you used or the full `api.deepgram.com` URL you sent your request to, including parameters.
- Any code snippets you can share.

**Comment by nikolawhallon** [comment, marked as answer]
TLDR we don't have a feature/flag for quite this exactly.

I would do this client-side, there are lots of opinions on how to transcribe spoken letters in different contexts. Imagine the transcript "yes my name is spelled n i k yeah and then o l a" if one were to just combine the letters it could read "yes my name is spelled nik yeah and then ola" (one can imagine this being more undesirable when spelling out a name like "Samwise").

Because it's hard to imagine a one-size-fits-all solution, having some logic in your client code to check the output word array for single characters, and stringing together multiple single-character words together (maybe if the sequence is 3 or more, to avoid false positives), might work well!

**Comment by diegoasua** [comment]
sounds good @nikolawhallon . I'll try that, thanks
